### DISCUSSION TITLE
Serverpod generate --watch Command Not Reflecting Endpoint Changes

### DISCUSSION THREAD

**Discussion by wanclem** [serverpod/serverpod#2049] [Q&A] [answered]
**Description**

When the Serverpod server is running and changes are made within endpoint classes, the `generate --watch` command does not seem to capture and reflect the latest modifications.

**Steps to Reproduce:**
1. Begin with a method defined within an endpoint class, for example:
```dart
class ExampleEndpoint extends Endpoint {
  Future<String> hello(Session session, String name) async {
    return 'Hello $name';
  }
}
```

Modify the method by changing the greeting keyword from "Hello" to "Welcome" as shown below and save:

```dart
class ExampleEndpoint extends Endpoint {
  Future<String> hello(Session session, String name) async {
    return 'Welcome $name';
  }
}
```

- Terminal Output: On the terminal, the following message is observed on save:
```less
File changed: modify lib/src/endpoints/example_endpoint.dart
✓ Generating code (73ms)
Incremental code generation complete.
```
Enter "Mike" as the name on the frontend and click the "Send to server" button.

Observe that despite the successful code generation, the response received continues to display "Hello Mike" instead of the expected "Welcome Mike".

**Expected behavior**
The `generate --watch` command should dynamically monitor and reflect any changes made within endpoint classes, ensuring that modifications to methods, such as the greeting keyword adjustment from "Hello" to "Welcome", are immediately propagated and applied to the server's behavior. This would guarantee consistency between frontend interactions and server responses.

**Platform information**
Android Emulator, Google Chrome browser

**Serverpod versions**
 -  _CLI Version:_ 1.2.5.
 - _Library version:_ 1.2.5.

**Comment by Isakdl** [comment, marked as answer]
Your specific example would not need the generate command to execute at all. Did you restart the server to get the latest changes?

**Comment by wanclem** [comment]
> Your specific example would not need the generate command to execute at all. Did you restart the server to get the latest changes?

I see. If that's the case, what's the command to restart the server and get the keyword changes propagated?

I want an auto-restart(classic  `nodemon` behavior in nodejs) whenever I make changes anywhere on the server. Stopping and restarting the server manually is a really tedious flow.

I'm not sure what I am doing wrong but even the watch

**Comment by Isakdl** [comment]
Currently, this functionality does not exist unfortunately but we intend to add it. 

The generate watch command will only make sure that your generated code is always updated, meaning the client library and various bindings on the server such as to the database, serialization, etc.

**Reply by wanclem** [reply]
I see. Thanks for the clarification! Looking forward to it!
